### ISSUE TITLE
org.influxdb.InfluxDBIOException: java.net.SocketException: No buffer space available (maximum connections reached?)

### ISSUE THREAD

**Issue by KDawgOne** [influxdata/influxdb-java#360]
While attempting to run multiple concurrent threads of the following:

while(true)
{
    		InfluxDB influxDB = InfluxDBFactory.connect("http://localhost:8086","","");
    		
    		BatchPoints batchPoints = BatchPoints
    				.database("mydb")
    				.consistency(ConsistencyLevel.ALL)
    				.build();    		
    		
    		Point point1 = Point.measurement("mydata")
					.time(Calendar.getInstance().getTimeInMillis(), TimeUnit.MILLISECONDS)
					.addField("value1",1234)
					.build();    

    		batchPoints.point(point1);
    		influxDB.write(batchPoints);
    		influxDB.close();
}

I get the error message:

org.influxdb.InfluxDBIOException: java.net.SocketException: No buffer space available (maximum connections reached?)

after a few seconds. The Microsoft Hotfix 2577795 "Kernel sockets leak on a multiprocessor computer that is running Windows Server 2008 R2 or Windows 7" is not the issue, since the newest version of WinSock is being run

Environment: Windows 7 SP1
InfluxDB version: 1.3.1
influsdb-java version: 2.7
JVM: Java 8
Application: Custom Java Client

netstat -a 

shows many ESTABLISHED connections, (None in CLOSE_WAIT).

**Comment by majst01** [comment]
Any specific reason why you dont  use the same influxDB connection ?

you code should look like:

```java
InfluxDB influxDB = InfluxDBFactory.connect("http://localhost:8086","","");

while(true) {
		BatchPoints batchPoints = BatchPoints
				.database("mydb")
				.consistency(ConsistencyLevel.ALL)
				.build();    		
		
		Point point1 = Point.measurement("mydata")
				.time(Calendar.getInstance().getTimeInMillis(), TimeUnit.MILLISECONDS)
				.addField("value1",1234)
				.build();    

		batchPoints.point(point1);
		influxDB.write(batchPoints);
}
influxDB.close();
```

**Comment by KDawgOne** [comment]
The example is part of a larger loop and the way the larger program works would make this approach impractical. I created the above example to highlight the issue.

I did find what seams as a 'solution' to this issue. It may be just a Windows issue. It has run for about 6 hours with no problems so far, whereas before it would fail after about 30 seconds. For anyone else who deals with this:

using regedt, locate the following key:

HKEY_LOCAL_MACHINE/SYSTEM/CurrentControlSet/services/Tcpip/Parameters

Add the following Field:

MaxUserPort, make it a DWORD and set it to 65534 (Dec)

Thanks for the response.
